chain/network/Cargo.toml

```diff
@@ -35,8 +35,10 @@ near-metrics = { path = "../../core/metrics" }
 near-performance-metrics = { path = "../../utils/near-performance-metrics" }
 near-performance-metrics-macros = { path = "../../utils/near-performance-metrics-macros" }
 near-stable-hasher = { path = "../../utils/near-stable-hasher" }
+near-framed-read = {path = "../../utils/near-framed-read" }
 
 delay-detector = { path = "../../tools/delay_detector", optional = true}
+pin-project-lite = "0.2.7"
```

Comment:
Removed

Reply:
Initially I intended this PR's code to be inside network, but I moved it out.

Reply:
`pin-project-lite` is a good dependency, in two senses: 

* it is small, well maintained crate
* it is a dependency of tokio itself, so we can't not use it

Meta heuristic: for dependencies, look at the changes in Cargo.**lock**. If there are new packages in lockfile, we need to think twice. If it just uses a package we use otherwise, it is probably OK (but still needs an "is this the right level of abstraction" sanity check.)